device/core/src/iotedge_authentication_provider.ts

```diff
@@ -123,36 +146,67 @@ export class IotEdgeAuthenticationProvider extends SharedAccessKeyAuthentication
-    this._restApiClient.executeApiCall('POST', path, null, signRequest, (err, body: SignResponse, response) => {
+    // Codes_SRS_NODE_IOTEDGED_AUTHENTICATION_PROVIDER_13_027: [** The `_sign` method shall use the `SharedAccessSignature.createWithSigningFunction` function to build the data buffer which is to be signed by iotedged.
+    SharedAccessSignature.createWithSigningFunction(this._credentials, expiry, (buffer, signCallback) => {
+      // Codes_SRS_NODE_IOTEDGED_AUTHENTICATION_PROVIDER_13_014: [ The _sign method shall build an object with the following schema as the HTTP request body as the sign request:
+      //   interface SignRequest {
+      //     keyId: string;
+      //     algo: string;
+      //     data: string;
+      //   }
+      //   ]
+
+      // Codes_SRS_NODE_IOTEDGED_AUTHENTICATION_PROVIDER_13_013: [ The _sign method shall build the sign request using the following values:
+      //   const signRequest = {
+      //     keyId: "primary"
+      //     algo: "HMACSHA256"
+      //     data: `${data}\n${expiry}`
+      //   };
+      //   ]
+      const signRequest: SignRequest = {
+        keyId: DEFAULT_KEY_ID,
+        algo: DEFAULT_SIGN_ALGORITHM,
+        // the data to be signed needs to have the expiry value appended with a newline
+        data: buffer.toString('base64')
+      };
+
+      // Codes_SRS_NODE_IOTEDGED_AUTHENTICATION_PROVIDER_13_019: [ The _sign method shall invoke this._restApiClient.executeApiCall to make the REST call on iotedged using the POST method. ]
+      // Codes_SRS_NODE_IOTEDGED_AUTHENTICATION_PROVIDER_13_025: [** The `_sign` method shall set the HTTP request option's `request` property to use the `http.request` object.
+      // Codes_SRS_NODE_IOTEDGED_AUTHENTICATION_PROVIDER_13_026: [** The `_sign` method shall set the HTTP request option's `port` property to use the workload URI's port if available.
+      this._restApiClient.executeApiCall(
+        'POST',
+        path,
+        { 'Content-Type': 'application/json' },
+        signRequest,
+        this.getRequestOptions(),
+        (err, body: SignResponse, response) => {
+          if (err) {
+            signCallback(err, null);
+          } else {
+            // Codes_SRS_NODE_IOTEDGED_AUTHENTICATION_PROVIDER_13_015: [ The _sign method shall invoke callback when the signature is available. ]
+            signCallback(null, Buffer.from(body.digest, 'base64'));
+          }
+        });
+    }, (err, sas) => {
       if (err) {
-        callback(err, null);
+        callback(err);
       } else {
-        // Codes_SRS_NODE_IOTEDGED_AUTHENTICATION_PROVIDER_13_015: [ The _sign method shall invoke callback when the signature is available. ]
-        callback(null, body.digest);
+        callback(null, sas.toString());
       }
     });
   }
+
+  private getRequestOptions(): HttpRequestOptions {
```

Comment:
>getRequestOptions [](start = 10, length = 17)

we prefix all private members with '_' (even if typescript has the private access modifier)

Reply:
Done